connectors/sources/sharepoint_server.py

```diff
@@ -410,20 +390,19 @@ async def _fetch_data_with_query(self, site_url, param_name):
             Response of the GET call.
         """
         skip = 0
+        response_result = []
```

Comment:
Why declare it here since you only do re-assigning on L402?

Reply:
On line 406 we are checking the length of response result for peroforming pagination over this, and if we remove from here then line 406 will give `response_result is not defined` error in case line 402 doesn't execute.